src/newrelic/processor_test.go

```diff
@@ -1139,3 +1158,155 @@ func TestShouldConnect(t *testing.T) {
+	//       So the actual value used to compare to the collector
+	//       limit will be 1/12th (5s/60s) smaller
+	var appInfo AppInfo = AppInfo{}
+	logHarvestLimit := 833
+
+	switch eventType {
+	case "log_event_data":
+		logHarvestLimit = int(collectorLimit)
+		appInfo = GetLogEventAppInfo(int(agentLimit))
+
+	default:
+		t.Fatalf("%s: runMockedCollectorHarvestLimitTest() invalid eventType \"%s\" specified", testName, eventType)
+	}
+
+	m := NewMockedProcessor(1)
+
+	id := AgentRunID("log")
+
+	reply := m.p.IncomingAppInfo(nil, &appInfo)
+	<-m.p.trackProgress // receive app info
+	if reply.State != AppStateUnknown {
+		t.Fatal("\"", testName, "\": expected ", AppStateUnknown, " got ", reply.State)
+	}
+
+	// mock collector response specifying non-zero log harvest limit
+	mockedReply := `{"agent_run_id":"` + id.String() +
+		`","zip":"zap",` +
+		`"span_event_harvest_config":{"report_period_ms":5000,"harvest_limit":166},` +
+		`"event_harvest_config":{"report_period_ms":5000,` +
+		`"harvest_limits":{"analytics_event_data":833,"custom_event_data":833,"error_event_data":833,` +
+		`"log_event_data":` + strconv.Itoa(logHarvestLimit) + `}}}`
+	m.DoConnectConfiguredReply(t, mockedReply)
+
+	reply = m.p.IncomingAppInfo(nil, &appInfo)
+	<-m.p.trackProgress // receive app info
+	if reply.State != AppStateConnected {
+		t.Fatal("\"", testName, "\": expected ", AppStateConnected, " got ", reply.State)
+	}
+
+	// should be 1 app in processor map and name should match that in appinfo
+	if 1 != len(m.p.apps) {
+		t.Error("\"", testName, "\": Exactly one application should be created.")
+	}
+
+	app, found := m.p.apps[appInfo.Key()]
+	if !found {
+		t.Error("\"", testName, "\":Expected application not found in processor map.")
+	}
+
+	// verify the client value was choosen
+	scaledAgentLimit := uint64(agentLimit / 12)
+	expectedLimit := uint64(collectorLimit)
+	if scaledAgentLimit < collectorLimit {
+		expectedLimit = scaledAgentLimit
+	}
+	finalLimit := app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit
+	if int(expectedLimit) != finalLimit {
+		t.Errorf("\" %s \": Expected %d to be choosen but %d was instead", testName, expectedLimit, finalLimit)
+	}
+
+	m.p.quit()
+}
+
+func TestConnectNegotiateLogEventLimits(t *testing.T) {
+
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 90*12, 100, "agent limit smaller than collector")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 110*12, 100, "agent limit larger than collector")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 100*12, 100, "agent limit equal to collector")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 0, 100, "agent limit == 0, collector != 0")
+	runMockedCollectorHarvestLimitTest(t, "log_event_data", 100*12, 0, "agent limit != 0, collector == 0")
+
+}
+
+func TestProcessLogEventLimit(t *testing.T) {
+	//cannedConnectReply := []byte(`{"agent_run_id":"1"}`) // parseConnectReply expects at least agent_run_id in collector reply
```

Comment:
```suggestion
```

Reply:
Good catch - fixed in commit [3bd6d67](https://github.com/newrelic/newrelic-php-agent/pull/497/commits/3bd6d67c7ab67a11c26ed97bc95b19d685d32509).